src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -382,6 +470,255 @@ public bool IsPackageDeprecated
+            }
+        }
+
+        private static bool IsHandleableBitmapEncodingException(Exception ex)
+        {
+            return ex is ArgumentException ||
+                ex is COMException ||
+                ex is FileFormatException ||
+                ex is InvalidOperationException ||
+                ex is NotSupportedException ||
+                ex is OutOfMemoryException ||
+                ex is IOException ||
+                ex is UnauthorizedAccessException;
+        }
+
+        private async Task FetchImageAsync(Uri iconUrl)
+        {
+            if (iconUrl is null)
+            {
+                return;
+            }
+
+            using (Stream stream = await GetStream(iconUrl))
+            {
+                if (stream != null)
+                {
+                    var iconBitmapImage = new BitmapImage();
+                    iconBitmapImage.BeginInit();
+                    iconBitmapImage.StreamSource = stream;
+
+                    try
+                    {
+                        FinalizeBitmapImage(iconBitmapImage);
+                        iconBitmapImage.Freeze();
+                        IconBitmap = iconBitmapImage;
+                        BitmapStatus = IconBitmapStatus.DownloadedIcon;
+                    }
+                    catch (Exception ex) when (IsHandleableBitmapEncodingException(ex))
+                    {
+                        IconBitmap = Images.DefaultPackageIcon;
+                        BitmapStatus = IconBitmapStatus.DefaultIconDueToDecodingError;
+                    }
+                }
+                else
+                {
+                    if (BitmapStatus == IconBitmapStatus.Fetching)
+                    {
+                        BitmapStatus = IconBitmapStatus.DefaultIconDueToNullStream;
+                    }
+                }
+
+                ErrorFloodGate.ReportAttempt();
+            }
+        }
+
+        private static void FinalizeBitmapImage(BitmapImage iconBitmapImage)
+        {
+            // Default cache policy: Per MSDN, satisfies a request for a resource either by using the cached copy of the resource or by sending a request
+            // for the resource to the server. The action taken is determined by the current cache policy and the age of the content in the cache.
+            // This is the cache level that should be used by most applications.
+            iconBitmapImage.UriCachePolicy = RequestCacheIfAvailable;
+
+            // Instead of scaling larger images and keeping larger image in memory, this makes it so we scale it down, and throw away the bigger image.
+            // Only need to set this on one dimension, to preserve aspect ratio
+            iconBitmapImage.DecodePixelWidth = DecodePixelWidth;
+
+            // Workaround for https://github.com/dotnet/wpf/issues/3503
+            iconBitmapImage.CreateOptions = BitmapCreateOptions.IgnoreColorProfile;
```

Comment:
FYI, this may no longer be needed, since your changes mean the underlying bug will no longer crash the VS process.

However, when I tried it locally, `IsHandleableBitmapEncodingException` does not handle the exception, so probably as @donnie-msft wrote in another comment, no icon ever loads, it just stays an empty space forever.

I still need to contact the WPF team to better understand the issue, and we can remove it in a future PR, but I'm trying to raise awareness.

Reply:
somebody added a test that loads an image that won't be able to load without that setting.
so i think it is better to keep this setting and show icons that need it, than go to default icons.